src/Microsoft.AspNet.Http.Abstractions/QueryString.cs

```diff
@@ -148,7 +148,7 @@ public static QueryString Create(IEnumerable<KeyValuePair<string, string>> param
         /// </summary>
         /// <param name="parameters"></param>
         /// <returns>The resulting QueryString</returns>
-        public static QueryString Create(IEnumerable<KeyValuePair<string, StringValues>> parameters)
+        public static QueryString Create(IQueryCollection parameters)
```

Comment:
Right, IQueryCollection implements IEnumerable<KeyValuePair<string, StringValues>>. The Response.Query setter is primarily used in unit tests so there's no need to add an strange public API to optimize it.